### DISCUSSION TITLE
New solaredge inline meter. the values look not right

### DISCUSSION THREAD

**Discussion by kokamoka** [binsentsu/home-assistant-solaredge-modbus#181] [General]
Hi Guys,

I just received my new solaredge home hub, with the new inline meter.
I have managed to get connect to the inverter, but some of the m1 values do not look right. The Inline meter modbus id is 2.
Hereby some screenshot of the values:
![image](https://github.com/binsentsu/home-assistant-solaredge-modbus/assets/303275/c4531078-06d5-4656-b0f3-4b4f0dcbf00a)
![image](https://github.com/binsentsu/home-assistant-solaredge-modbus/assets/303275/8cfb80d6-1af3-4ba0-b628-098edd5e2080)
![image](https://github.com/binsentsu/home-assistant-solaredge-modbus/assets/303275/86cb94c3-f125-44ac-a96e-7343188b92f1)
![image](https://github.com/binsentsu/home-assistant-solaredge-modbus/assets/303275/d71e742d-4c9c-4052-8f7f-d0d15a38f429)

Inline meter:
![image](https://github.com/binsentsu/home-assistant-solaredge-modbus/assets/303275/62eaac0d-5082-4aee-8893-643b58330cc2)
https://www.solaredge.com/nl/products/solaredge-inline-meter

The configuration is:
 {
        "entry_id": "xxxxxxxxxxxxxxxxxxxxxxx",
        "version": 1,
        "domain": "solaredge_modbus",
        "title": "SolarEdgeModbus",
        "data": {
          "name": "SolarEdgeModbus",
          "host": "x.x.x.X",
          "port": 1502,
          "modbus_address": 1,
          "power_control": false,
          "read_meter_1": true,
          "read_meter_2": false,
          "read_meter_3": false,
          "read_battery_1": false,
          "read_battery_2": false,
          "scan_interval": 10
        },
        "options": {},
        "pref_disable_new_entities": false,
        "pref_disable_polling": false,
        "source": "user",
        "unique_id": "x.x.x.x",
        "disabled_by": null
      }

Is there any way i can fix it?
Thanks in advance.

**Comment by ChristophCaina** [comment]
what values don't look right?
Do you have any comparision between the values the meter itself is showing?

**Comment by miqm** [comment]
not sure, but this might be related: https://github.com/binsentsu/home-assistant-solaredge-modbus/discussions/215#discussion-5956647 - I've problems with meter integrated in backup box - AC Power seems ok, but KWh are totally odd.

**Reply by miqm** [reply]
TBH this seems like some misread from modbus registries or change in specification how the data is exposed (if it is exposed at all). 65535 is the maximum value of unsigned short represented binary as `1111111111111111`. can't be a coincidence .

**Reply by miqm** [reply]
I also did a check with a different solaredge modbus reader (EnergyExporter) and it reports this same values as this integration. So I assume SolarEdge broke something on their side and does not fill data correctly on TCP modbus.

**Reply by miqm** [reply]
For me problem has been solved by updating the firmware.

**Reply by SimonX200** [reply]
Have the same problem with the backupbox since installed in October. Energy values are too low by 30%. Noticed that in the mySolar App. Level1 could not help. Inline Meter before was ok'ish. Have a separate house consumption meter (shelly pro 3em) since July. Started to use HA this week.
